packages/logger/src/formatters/ConsoleFormatter.ts

```diff
@@ -0,0 +1,77 @@
+import chalk from 'chalk';
+import {info, warning, error} from 'log-symbols';
+import prettyMs from 'pretty-ms';
+
+import {LogLevel, FormatEntry, Formatter} from '../types';
+
+export class ConsoleFormatter implements Formatter {
+  private startTime: [number, number];
+
+  constructor() {
+    this.startTime = process.hrtime();
+  }
+
+  next(entry: FormatEntry) {
+    const timeDiff = process.hrtime(this.startTime);
+    const msDiff = timeDiff[0] * 1000 + timeDiff[1] / 1000;
+    const timestamp = chalk.magenta(`+${prettyMs(msDiff)}`);
+
+    const {level, payload, scopes} = entry;
+
+    let logMsg = `${payload} ${timestamp}`;
+
+    if (level === LogLevel.Critical) {
+      const {message, stack} = payload as Error;
+      logMsg = message;
+
+      if (stack != null) {
+        const [name, ...lines] = stack.split('\n');
+        logMsg = `${name} ${timestamp}\n${lines
```

Comment:
So if there's a stack we overwrite the whole log message with the stack and add a timestamp? I'm not sure the intention here.

Reply:
(ie we won't see the error's `message`)

Reply:
Sorry, I agree thsi is a bit confusing. `name` is probably the wrong identifier here. The first line of the stack trace contains the error type and message. Also, I wanted the timestamp to appear ont he first line, alongside the message, rather than after the stack trace.